tests/e2e/etcd_config_test.go

```diff
@@ -47,11 +47,7 @@ func TestEtcdMultiPeer(t *testing.T) {
 	peers, tmpdirs := make([]string, 3), make([]string, 3)
 	for i := range peers {
 		peers[i] = fmt.Sprintf("e%d=http://127.0.0.1:%d", i, e2e.EtcdProcessBasePort+i)
-		d, err := os.MkdirTemp("", fmt.Sprintf("e%d.etcd", i))
-		if err != nil {
-			t.Fatal(err)
-		}
-		tmpdirs[i] = d
+		tmpdirs[i] = t.TempDir()
```

Comment:
In such situation where we have meaningful identifier in the directory name that points on the instance of etcd from the test I would prefer to preserve it in the path. 

```
rootTmpDir=t.TempDir()
	for i := range peers {
	  d, err := os.MkdirTemp(rootTmpDir, fmt.Sprintf("%d-%d.etcd", i, e2e.EtcdProcessBasePort+i))
	}
```
This makes it easier to correlate logs (and path in the logs) with a directory that might contain a corrupted state.